src/randomizer.py

```diff
@@ -2080,6 +2103,85 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+def perm_inline_get_structmember(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Creates an inline function for accessing a struct member."""
+
+    typemap = build_typemap(ast)
+
+    cands: List[ca.StructRef] = []
+
+    class Visitor(ca.NodeVisitor):
+        def visit_StructRef(self, node: ca.StructRef) -> None:
+            if (
+                region.contains_node(node)
+                and isinstance(node.name, ca.ID)
+                and not ast_util.is_assignment_lvalue(fn.body, node)
+            ):
+                cands.append(node)
+            self.generic_visit(node)
+
+    Visitor().visit(fn.body)
+    ensure(cands)
+
+    cand = random.choice(cands)
+
+    parentOp: Optional[ca.UnaryOp] = ast_util.find_parent_addressOP(fn.body, cand)
+
+    member_type: SimpleType = decayed_expr_type(cand, typemap)
+    member_name = cand.field.name
+
+    new_inline_fn_name = get_noncolliding_fn_name(ast, f"get_{member_name}")
+
+    # Replace the StructRef with a FuncCall to the new inline yet to be created
+    replace_node(
+        fn.body,
+        cand if not parentOp else parentOp,
+        ca.FuncCall(ca.ID(new_inline_fn_name), ca.ExprList([cand.name])),
+    )
+
+    # Now create the new inline function definition
+    arg_type: SimpleType = decayed_expr_type(cand.name, typemap)
+    assert isinstance(arg_type.type, ca.TypeDecl)
+    assert arg_type.type.declname
```

Comment:
I don't think this is true, and you shouldn't need it

Reply:
So, this is for magically picking a parameter name out of thin air. When `cand.name` is a ca.ID, which it previously always was, `expr_type` would find its declaration and return that Decl's type, which would incidentally have a name set on it (because of the weird way in which the AST works for declarations) which matches the ca.ID's name. This isn't something that `expr_type` guarantees will work though, it's very subtle, and it will break for non-ca.ID's... Instead I would suggest creating a custom parameter name (say, `"x"`, or falling back to cand.name.name if cand.name is an ca.ID if you want).